src/tools/collectors.rs

```diff
@@ -0,0 +1,171 @@
+use std::collections::BTreeSet;
+
+use super::prelude::*;
+
+/// Collect data about blob
+#[derive(Debug)]
+pub struct BlobSummaryCollector {
+    header: BlobHeader,
+    records: usize,
+    deleted_records: usize,
```

Comment:
You should expose new values through access functions in the same way the other values exposed below